sdk/core/core/src/az_http_request.c

```diff
@@ -34,7 +34,8 @@ AZ_NODISCARD az_result az_http_request_init(
   AZ_PRECONDITION_VALID_SPAN(headers_buffer, 0, false);
 
   int32_t query_start = 0;
-  az_result url_with_query = _az_scan_until(url, _az_is_question_mark, &query_start);
+  az_result url_with_query = _az_span_scan_until(
+      az_span_init(az_span_ptr(url), url_length), _az_is_question_mark, &query_start);
```

Comment:
Not span_init, can you use az_span_slice(url, url_length))?

Reply:
az_span_slice calls az_span_init.
slice only checks range and indexes before calling az_span_init.

At the point of creating an HTTP Request, an SDK client has gotten some url from customer and it has allocated in stack a buffer to hold that url plus anything that gets added to it. It has validated the size and got the lenght of url.

Making http request checking again the range looks duplicating the same validations.

Reply:
this is basically just calling what it used to be az_span_builder_result(...)   where we don't need to validate if the length is still inside the az_span size